- Added ability to overwrite the options in the field completely

ajvwhite opened the pull request:
When using the ORM and the fieldset generation based on the Model, I
have found sometimes it is necessary to overwrite a select field's
options after being initialised by an init function completely
(different load conditions for the select perhaps based on a user action).

This small tweak will allow that, but the default behaviour stays the
same.

ajvwhite committed:
- Added ability to overwrite the options in the field completely

When using the ORM and the fieldset generation based on the Model, I
have found sometimes it is necessary to overwrite a select field's
options completely (different load conditions for the select perhaps
based on a user action).

This small tweak will allow that, but the default behaviour stays the
same.

huglester commented on classes/fieldset/field.php:
```diff
@@ -361,7 +362,7 @@ public function set_options($value, $label = null)
 		};
 
-		empty($this->options) ? $this->options = $value : $merge($this->options, $value, $merge);
+		($replace_options || empty($this->options)) ? $this->options = $value : $merge($this->options, $value, $merge);
```
should be $replace_options or empty($this->options)
according to coding standards

ajvwhite replied:
Wasn't sure if that preference was an absolute requirement, but thank you for pointing it out, has been amended to match to the stipulated standards

ajvwhite committed:
Change to || comparison operator to or

Change of the comparison operator in the last tweak to fit in fit
FuelPHP coding standard preferences. Also fixed a typo in the block
comment
